server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1397,6 +1377,63 @@ public boolean performConsumerUpdates(ConsumerDTO updated, Consumer toUpdate,
         return changesMade;
     }
 
+    private boolean updateEnvironment(ConsumerDTO updated, Consumer toUpdate) {
+        boolean changesMade = false;
+        boolean environmentFound = false;
+        Environment e = null;
+        String environmentId = updated.getEnvironment() == null ? null : updated.getEnvironment().getId();
+        String environmentName = updated.getEnvironment() == null ? null : updated.getEnvironment().getName();
+
+        if (environmentName != null && environmentId == null) {
+            List<Environment> environment = environmentCurator.listForOwnerByName(toUpdate.getOwner(),
+                environmentName).list();
+
+            if (!environment.isEmpty() && environment.size() == 1) {
```

Comment:
This is a redundant check. If the environment collection size is 1, we know it's not empty. Since we're not differentiating between empty and more-than-one in the error output, we can simplify this down to just the size check.

Reply:
yes sure.